### PR TITLE
Content Baseline Anchors

### LINKED ISSUE #683: [Feature Request] Content-Relative Anchors
Cetz currently doesn't support anchors whose position is relative to the properties of a nodes content—see `base*`, `mid*`, and `text` in the image below.

grafik
(Credits to [Packt](https://web.archive.org/web/20240813122622/https://subscription.packtpub.com/book/business-and-other/9781804618233/3/ch03lvl1sec22/using-shapes-and-anchors) for the image.)

After a short, productive discussion with @fenjalien (see #680), it turns out that:
> No we currently don't [support] as it would require inspecting the provided content and measuring it. This is difficult as theres no official documentation for how content values are supposed to act and could change with any release. Once Typst provides more ways to control content it would be something we'll be looking into.
<sup>— Originally commented by **fenjalien** in the [discussion](https://github.com/cetz-package/cetz/discussions/680#discussioncomment-10325260) mentioned above.</sup>

### PR DESCRIPTION
**Pull request by johannes-wolf**
Fixes #683

Lets add this to 0.3.

### CODE REVIEW

File: src/draw/shapes.typ [hunk 1]
```diff
@@ -788,47 +794,82 @@
...
+        base-east: base-east,
+        base-west: base-west,
+        north: north,
+        north-east: north-east,
+        north-west: north-west,
+        south: south,
+        south-east: south-east,
+        south-west: south-west,
+        east: east,
+        west: west,
```
Comment (timfi) [inline src/draw/shapes.typ:872]: Very cool to see this added already! As far as can see, this PR currently only adds `base-*` and `mid-*` anchors. All that remains to achieve parity with Tikz in this regard is the `text` anchor, which lies where the contents baseline and western edge meet.

File: src/draw/shapes.typ [hunk 2]
```diff
@@ -736,6 +736,13 @@
 ///
 /// ## Anchors
 /// Supports border anchors.
+/// - **mid**: Content center, from baseline to top bounds
+/// - **mid-east**: Content center extended to the east
+/// - **mid-west**: Content center extended to the west
+/// - **base**: Horizontally centered baseline of the content
+/// - **base-east**: Baseline height extended to the east
+/// - **base-west**: Baseline height extended to the west
+/// - **text**: Position at the content start on the baseline of the content
```
Comment (fenjalien) [inline src/draw/shapes.typ:745]: It should be stated which anchor is now the default

**Review by fenjalien — COMMENTED** [review]
Can we merge this onto master instead of the release branch so the commit doesn't get squashed?

**Review by fenjalien — COMMENTED** [review]
I think you missed some changes when changing branches, the gallery and docs folders shouldn't have any changes...

**Comment by johannes-wolf** [comment]
Was containing the changes for the 0.3 release. I've removed them.

**Review by fenjalien — APPROVED** [review]

**Pull request merged**